conda/core/initialize.py

```diff
@@ -439,10 +454,28 @@ def make_initialize_plan(conda_prefix, shells, for_user, for_system, anaconda_pr
             raise NotImplementedError()
 
     if 'powershell' in shells:
-        if for_user:
-            raise NotImplementedError()
+        # There's several places PowerShell can store its path, depending
+        # on if it's Windows PowerShell, PowerShell Core on Windows, or
+        # PowerShell Core on macOS/Linux. The easiest way to resolve it is to
+        # just ask different possible installations of PowerShell where their
+        # profiles are.
+        config_powershell_paths = list(
+            map(find_powershell_path, ['powershell', 'pwsh', 'pwsh-preview'])
+        )
```

Comment:
Let's pull `find_powershell_path()` in as an inner function here.  Since it's subprocessing, I'd rather have it more hidden, until we explicitly need it somewhere else.  And then how about this instead...

```python
def find_powershell_paths(*exe_names):
    for exe_name in exe_names:
        try:
            yield subprocess_call((exe_name, '-NoProfile', '-Command', '$PROFILE')).stdout.strip()
        except Exception:
            pass

config_powershell_paths = tuple(find_powershell_paths('powershell', 'pwsh', 'pwsh-preview'))
```

Couple notes.  LGTM yells at us if we use bare `except:`.  It ends up catching lower-level exceptions than you usually want.  And throughout the conda codebase we always prefer tuples to lists by default, unless we explicitly need the mutability.

Reply:
Sounds good, I've replaced with the code you suggested, modulo some minor reformatting to avoid the long lines flake8 rule.